libra-node/src/main_node.rs

```diff
@@ -126,16 +126,32 @@ pub fn setup_network(
         let seed_peers = config.seed_peers.seed_peers.clone();
         let signing_private = config
             .network_keypairs
+            .as_mut()
+            .unwrap()
             .signing_keys
             .take_private()
             .expect("Failed to take Network signing private key, key absent or already read");
-        let signing_public = config.network_keypairs.signing_keys.public().clone();
+        let signing_public = config
+            .network_keypairs
```

Comment:
recommend either the above approach or deriving the public from the private

Reply:
Thanks. Done -- please see above.